data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/ConvertEntryTypeProcessor.java

```diff
@@ -80,7 +83,18 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     if (keyVal != null) {
                         if (!nullValues.contains(keyVal.toString())) {
                             try {
-                                recordEvent.put(key, converter.convert(keyVal, converterArguments));
+                                if (keyVal instanceof ArrayList || keyVal.getClass().isArray()) {
+                                    ArrayList<Object> inputList;
```

Comment:
should probably be List<> here

Reply:
Do we need to support other list types? Since this is supporting "arrays", I thought "ArrayList" and not generic "List" is better.